### PR TITLE
Fix false positive duplicate static products lint rule

### PR DESCRIPTION
**Pull request by kwridan**
Resolves https://github.com/tuist/tuist/issues/2197

### Short description 📝

The duplicate static products linter was incorrectly flagging duplicate static products between host app targets and their additional bundled products (such as watch apps, app clips, extensions, etc...).

Those additional targets can safely link the same static products as their host app due them being independent products.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] A developer other than the author has verified that the changes work as expected.
- [x] The changes have been tested following the [documented guidelines](https://tuist.io/docs/contribution/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [ ] ~In case the PR introduces changes that affect users, the documentation has been updated~.

### Test Plan 🛠

- Run `swift build && swift run tuist generate --path fixtures/ios_app_with_watchapp2/`
- Verify no duplicate static products lint warning is produced

### CODE REVIEW

File: fixtures/ios_app_with_watchapp2/Derived/Sources/Bundle+WatchAppExtension.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// swiftlint:disable all
+import Foundation
```
Comment (pepibumur) [inline fixtures/ios_app_with_watchapp2/Derived/Sources/Bundle+WatchAppExtension.swift:2]: I'd add the derived folder to the .gitignore

**Review by pepibumur — APPROVED** [review]
Thanks for tackling this so quickly @kwridan

**Comment by kwridan** [comment]
I've had to update the acceptance test build commands (the ones in `xcode.rb`) as explicitly specifying `-sdk` when building schemes that contain multiple products of different platforms (like in the fixture I modified) can yield build errors. Instead the destination platform is specified to mitigate this. Did we encounter issues in the past that required explicitly specifying `-sdk`?

I did a sanity check on a few acceptance tests and that seemed to have worked, I guess we'll see how all of them fair on CI 😅 - if that fails, will revert the fixture change or add a different one not run by acceptance tests :/

**Comment by kwridan** [comment]
looks like the change works for all fixtures on CI too 🎉 - the only CI failure is `Rubocop` but those look unrelated to this change :/ 

Screenshot 2020-12-27 at 10 24 39

**Pull request merged**